packages/tools/devtools/devtools-view/src/components/data-visualization/EditableView.tsx

```diff
@@ -0,0 +1,384 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import React from "react";
+
+import {
+	Button,
+	Dropdown,
+	DropdownProps,
+	Input,
+	InputOnChangeData,
+	Option,
+	tokens,
+} from "@fluentui/react-components";
+import { DataEdit, FluidObjectValueNode, Primitive } from "@fluid-experimental/devtools-core";
+
+import { Serializable } from "@fluidframework/datastore-definitions";
+import { useMessageRelay } from "../../MessageRelayContext";
+import { TreeHeader } from "./TreeHeader";
+
+/**
+ * Input to {@link EditableValueView}
+ */
+export interface EditableViewProps {
+	node: FluidObjectValueNode;
+	containerKey: string;
+	label: string;
+}
+
+/**
+ * Render data that is Editable with its corresponding UI
+ *
+ * @remarks {@link ContainerFeaturesContext} must be set in order to use this component.
+ */
+export function EditableView(props: EditableViewProps): React.ReactElement {
+	const { node, containerKey, label } = props;
+
+	/**
+	 * State to store whether the user is currently editing data
+	 */
+	const [editing, setEditing] = React.useState<boolean>(false);
+
+	/**
+	 * State to store the current type, or the type of the intended edit
+	 */
+	const [type, setType] = React.useState<string>(typeof node.value);
+
+	/**
+	 * State to store the corresponding UI component
+	 */
+	const [editingComponent, setEditingComponent] = React.useState<React.ReactElement>(<></>);
```

Comment:
Is this necessary? Can we instead derive which component to use via `type`?

Reply:
I was thinking I can use the selectEditUI function that gets used by everything else. However the organization of the code is a bit gross since I have to move that function and depedencies above the stateuse line

Reply:
I'm not sure we need anything that complex. Unless I'm missing something, we can do something like the following:

```typescript

let innerView: React.ReactElement;
switch(type):
{
  case "number":
    innerView = <NumberEditor .../>;
    break;
  ...
}
```
return (
  <div>
    ...
    {innerView}
  </div>
);

Reply:
We should chat in a call about this, I'm not sure I can explain through text adequately.